warehouse/warehouse.go

```diff
@@ -768,23 +848,9 @@ func (wh *HandleT) mainLoop() {
 				continue
 			}
 			setDestInProgress(warehouse, true)
-
-			_, ok := inRecoveryMap[warehouse.Destination.ID]
-			if ok {
-				whManager, err := warehousemanager.NewWhManager(wh.destType)
-				if err != nil {
-					panic(err)
-				}
-				logger.Infof("WH: Crash recovering for %s:%s", wh.destType, warehouse.Destination.ID)
-				err = whManager.CrashRecover(warehouseutils.ConfigT{
-					DbHandle:  wh.dbHandle,
-					Warehouse: warehouse,
-				})
-				if err != nil {
-					setDestInProgress(warehouse, false)
-					continue
-				}
-				delete(inRecoveryMap, warehouse.Destination.ID)
+			if err := wh.handleCrashRecovery(warehouse); err != nil {
+				setDestInProgress(warehouse, false)
```

Comment:
Can you move the error log here from inside the function? Otherwise, it looks like we completely ignore the error.